packages/circuit-ui/components/Modal/Modal.tsx

```diff
@@ -171,113 +165,106 @@ export const Modal: ModalComponent<ModalProps> = ({
+              outline: none;
+              background-color: ${theme.colors.white};
 
               &::after {
-                height: ${theme.spacings.mega};
+                position: fixed;
+                content: '';
+                display: block;
+                right: 0;
+                bottom: 0;
+                left: 0;
+                background: linear-gradient(
+                  rgba(255,255,255,0),
+                  rgba(255,255,255,0.66),
+                  rgba(255,255,255,1)
+                );
               }
-            }
 
-            ${theme.mq.kilo} {
-              top: 50%;
-              left: 50%;
-              transform: translate(-50%, -50%);
-              opacity: 0;
-              transition: opacity ${TRANSITION_DURATION_DESKTOP}ms ease-in-out;
-              border-radius: ${theme.borderRadius.mega};
+              ${theme.mq.untilKilo} {
+                right: 0;
+                bottom: 0;
+                left: 0;
+                transform: translateY(100%);
+                transition: transform ${TRANSITION_DURATION_MOBILE}ms ease-in-out;
 
-              &::after {
-                height: ${theme.spacings.giga};
-                border-bottom-left-radius: ${theme.borderRadius.mega};
-                border-bottom-right-radius: ${theme.borderRadius.mega};
+                &::after {
+                  height: ${theme.spacings.mega};
+                }
+              }
+
+              ${theme.mq.kilo} {
+                top: 50%;
+                left: 50%;
+                transform: translate(-50%, -50%);
+                opacity: 0;
+                transition: opacity ${TRANSITION_DURATION_DESKTOP}ms ease-in-out;
+                border-radius: ${theme.borderRadius.mega};
+
+                &::after {
+                  height: ${theme.spacings.giga};
+                  border-bottom-left-radius: ${theme.borderRadius.mega};
+                  border-bottom-right-radius: ${theme.borderRadius.mega};
+                }
               }
-            }
           `,
             variant === 'contextual' &&
               cssString`
-              ${theme.mq.untilKilo} {
-                border-top-left-radius: ${theme.borderRadius.mega};
-                border-top-right-radius: ${theme.borderRadius.mega};
-              }
-            `,
-            variant === 'immersive' &&
-              cssString`
-              /* iOS viewport bug fix */
-              /* https://allthingssmitty.com/2020/05/11/css-fix-for-100vh-in-mobile-webkit/ */
-              @supports (max-height: -webkit-fill-available) {
                 ${theme.mq.untilKilo} {
                   border-top-left-radius: ${theme.borderRadius.mega};
                   border-top-right-radius: ${theme.borderRadius.mega};
                 }
-              }
+            `,
+            variant === 'immersive' &&
+              cssString`
+                top: 0;
```

Comment:
@robinmetral This did the trick for fullscreen on iOS: by setting both `top` and `bottom` to `0`, we can use `100%` instead of `100vh`.